tests/as_mut.rs

```diff
@@ -1,117 +1,527 @@
+                first: "test".to_owned(),
+            };
+
+            assert!(ptr::eq(item.as_mut(), &mut item.first));
+        }
+
+        #[derive(AsMut)]
+        struct FieldForward {
+            #[as_mut(forward)]
+            first: String,
+        }
+
+        #[test]
+        fn field_forward() {
+            let mut item = FieldForward {
+                first: "test".to_owned(),
+            };
+
+            let rf: &mut str = item.as_mut();
+            assert!(ptr::eq(rf, item.first.as_mut()));
+        }
+
+        mod generic {
+            use super::*;
+
+            #[derive(AsMut)]
+            struct NoAttr<T> {
+                first: T,
+            }
+
+            #[test]
+            fn no_attr() {
+                let mut item = NoAttr {
+                    first: "test".to_owned(),
+                };
+
+                assert!(ptr::eq(item.as_mut(), &mut item.first));
+            }
+
+            #[derive(AsMut)]
+            #[as_mut(forward)]
+            struct StructForward<T> {
+                first: T,
+            }
+
+            #[test]
+            fn struct_forward() {
+                let mut item = StructForward {
+                    first: "test".to_owned(),
+                };
+
+                let rf: &mut str = item.as_mut();
+                assert!(ptr::eq(rf, item.first.as_mut()));
+            }
+
+            #[derive(AsMut)]
+            struct FieldEmpty<T> {
+                #[as_mut]
+                first: T,
+            }
+
+            #[test]
+            fn field_empty() {
+                let mut item = FieldEmpty {
+                    first: "test".to_owned(),
+                };
+
+                assert!(ptr::eq(item.as_mut(), &mut item.first));
+            }
+
+            #[derive(AsMut)]
+            struct FieldForward<T> {
+                #[as_mut(forward)]
+                first: T,
+            }
+
+            #[test]
+            fn field_forward() {
+                let mut item = FieldForward {
+                    first: "test".to_owned(),
+                };
+
+                let rf: &mut str = item.as_mut();
+                assert!(ptr::eq(rf, item.first.as_mut()));
+            }
+        }
     }
 }
 
-#[derive(AsMut)]
-struct SingleFieldGenericStruct<T> {
-    first: T,
-}
+mod multi_field {
+    use super::*;
+
+    mod tuple {
+        use super::*;
+
+        #[derive(AsMut)]
+        struct NoAttr(String, i32);
+
+        #[test]
+        fn no_attr() {
+            let mut item = NoAttr("test".to_owned(), 0);
+
+            assert!(ptr::eq(item.as_mut(), &mut item.0));
+            assert!(ptr::eq(item.as_mut(), &mut item.1));
+        }
 
-#[test]
-fn single_field_generic_struct() {
-    let mut item = SingleFieldGenericStruct { first: "test" };
+        #[derive(AsMut)]
+        struct Skip(String, i32, #[as_mut(skip)] f64);
 
-    assert!(ptr::eq(&mut item.first, item.as_mut()));
+        impl AsMut<f64> for Skip {
```

Comment:
@MegaBluejay it's a good practice to place here a comment why this impl is here, so the people don't wonder what the heck is that, and don't ask redundant questions.